txbridge/src/main/java/org/jboss/jbossts/txbridge/outbound/OutboundBridgeManager.java

```diff
@@ -66,18 +69,21 @@
 
         try
         {
-            Transaction transaction = (Transaction)TransactionManager.transactionManager().getTransaction();
+            Transaction transaction = (Transaction)TransactionManager.transactionManager(new InitialContext()).getTransaction();
 
-            Uid externalTxId = transaction.get_uid();
+            // generating new Uid to get it connected to incoming transaction
+            // the incoming transaction could come from different TM than Narayana is (e.g. Wildfly Transaction client)
+            transctionToUidMappings.putIfAbsent(transaction, new Uid());
```

Comment:
the trouble is that I'm not able to get `transaction.get_uid()` as there is used `javax.transaction.Transaction` interface which is returned from wfly transaction client. That way I can't take the uid from transaction directly. Because of it there is generated new `Uid` which is bound to incoming transaction where I rely on the transaction equality (as you mentioned) and generating new `Uid` which is needed for later recovery usage (Uid is saved to txn log).